amqp/src/main/scala/akka/stream/alpakka/amqp/AmqpConnectionSettings.scala

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.amqp
+
+import com.rabbitmq.client.{Address, Connection, ConnectionFactory, ExceptionHandler}
+
+/**
+ * Only for internal implementations
+ */
+sealed trait AmqpConnectionSettings {
+  private[amqp] def getConnection: Connection
+}
+
+/**
+ * Connects to a local AMQP broker at the default port with no password.
+ */
+case class AmqpConnectionLocal() extends AmqpConnectionSettings {
+  override def getConnection: Connection = new ConnectionFactory().newConnection
+}
```

Comment:
No need for a class, use a single `object` extending `AmqpConnectionSettings` instead.